numpy/core/src/multiarray/nditer_constr.c

```diff
@@ -1609,61 +1669,58 @@ npyiter_fill_axisdata(NpyIter *iter, npy_uint32 flags, npyiter_opitflags *op_itf
             }
             else {
                 int *axes = op_axes[iop];
-                int i = axes[ndim-idim-1];
-                if (i >= 0) {
-                    if (bshape == 1 || op_cur == NULL) {
+                npy_bool reduction_axis;
```

Comment:
nit: feel free to ignore. would be good to have some NPY_IT_DBG_PRINT in npyiter_fill_axis_data, since it hardly outputs anything if it touches this code path, except probably iter shape.